Ddr driver updates

akashdeep-ti opened the pull request:
DDR lpm stub driver cleanup. 
Add support to successfully resume from deep sleep or RTC + DDR low power mode from both frequency set points.

akashdeep-ti committed:
PENDING: fix(ti): Cleanup the LPM DDR driver

Cleanup the LPM DDR driver to remove the macros that are redundant.
Remove the includes that are not required.

Fixes: c7afea5543ff("PENDING: feat(ti): lpm: Add LPM low level drivers")
Change-Id: I19891295577036de62a7ba5c84a262ce70a46ac8
Signed-off-by: Akashdeep Kaur <a-kaur@ti.com>

akashdeep-ti committed:
PENDING: feat(ti): Add LPM support for two FSPs

Update the DDR driver in LPM so that the PHY registers for both FSP sets
are saved and restored in deep sleep low power mode. Also, ensure that
the init frequency is updated to current frequency set.
Set the valid frequency register during rtc plus ddr mode to ensure DDR
memory enters lpm and resumes from it.

Change-Id: I379e96d02a8b709d11bfcb0acefc44b77a9462a8
Signed-off-by: Akashdeep Kaur <a-kaur@ti.com>

akashdeep-ti commented:
Tested deep sleep and RTC + DDR mode to be working with these changes. http://serenity.dal.design.ti.com:7777/nidawasela.yaml

ti-sebin commented on plat/ti/k3/board/am62l/lpm/ddr.c:
```diff
@@ -7,89 +7,61 @@
+/* Register block core (PHY_1280-PHY_1405) offset and total */
+#define DDRSS_PHY_CORE_REGISTER_BLOCK_OFFS		0x5400U
+#define NUM_DDR_PHY_REG					126U
+#define NUM_ALL_PHY_REG					(NUM_DDR_DATA_0_REG + NUM_DDR_DATA_1_REG + NUM_DDR_ADDR_0_REG + NUM_DDR_ADDR_1_REG + NUM_DDR_ADDR_2_REG + NUM_DDR_PHY_REG)
```
Should we split it into multiple lines?

akashdeep-ti replied:
yes, updated! thanks

glneo commented on plat/ti/k3/board/am62l/lpm/ddr.c:
```diff
@@ -305,6 +355,39 @@ __wkupsramfunc void restore_ddr_registers(struct emif_handle_s *h)
 	}
+
+	/* Restore the second frequency set conditionally */
+	if ( ddrss_is_fsp_supported == 1) {
```
Random space after opening `(`, please run the code formatter.

DhruvaG2000 replied:
Hmm curious how checkpatch didn't catch this though 
https://github.com/TexasInstruments/arm-trusted-firmware/actions/runs/19894059078/job/57019810534?pr=58

akashdeep-ti replied:
Updated. Now, ran the code formatter. Thanks!!

glneo commented:
`DDR` and `LPM` are acronyms, capitalize them everywhere in your comments and commit message.

DhruvaG2000 commented on plat/ti/k3/board/am62l/lpm/ddr.c:
```diff
@@ -44,13 +44,18 @@
+#define DDRSS_PHY_CORE_REGISTER_1281_POS		0x1U
+#define DDRSS_PHY_CORE_REGISTER_1281_MULTICAST_EN	BIT(8)
+#define DDRSS_PHY_CORE_REGISTER_1281_FREQ_SEL_INDEX	BIT(16)
 #define NUM_ALL_PHY_REG					(NUM_DDR_DATA_0_REG + NUM_DDR_DATA_1_REG + NUM_DDR_ADDR_0_REG \
```
The way the commits have been split seems a bit tricky to me, because this macro for eg. is actually used in the commit  ("PENDING: feat(ti): Add lpm support for two FSPs") but is introduced in the other "cleanup" commit.
So essentially this will be a big challenge if we want to cherrypick this feature particularly to some upstream or different baseline. Let's try to partition changes carefully even though it's an SDK branch.

akashdeep-ti replied:
This change is added in the FSP commit and not cleanup commit

DhruvaG2000 replied:
okay, I may have confused it earlier.

DhruvaG2000 commented on plat/ti/k3/board/am62l/lpm/ddr.c:
```diff
@@ -7,89 +7,62 @@
+/* Register block core (PHY_1280-PHY_1405) offset and total */
+#define DDRSS_PHY_CORE_REGISTER_BLOCK_OFFS		0x5400U
+#define NUM_DDR_PHY_REG					126U
+#define NUM_ALL_PHY_REG					(NUM_DDR_DATA_0_REG + NUM_DDR_DATA_1_REG + NUM_DDR_ADDR_0_REG \
```
Please let this cleanup commit be only specific to cleanups. Let's try not adding new macros here that will be used at some later point. it will all be buried in huge lines of change and becomes harder to review.

akashdeep-ti replied:
Updated. Thanks

akashdeep-ti committed:
PENDING: fix(ti): Update the DDR FSP change sequence

Update the sequence used to change the frequency set point to fix
invalid req type masking and add the complete sequence.
Move it to a function so that it can be used to change FSP as required.

Fixes: c7afea5543ff("PENDING: feat(ti): lpm: Add LPM low level drivers")
Change-Id: I2ead4512698896c6b7011881d534ec9eb0419beb
Signed-off-by: Akashdeep Kaur <a-kaur@ti.com>

akashdeep-ti commented:
> `DDR` and `LPM` are acronyms, capitalize them everywhere in your comments and commit message.

Does this also apply to variable names?

glneo commented:
> > `DDR` and `LPM` are acronyms, capitalize them everywhere in your comments and commit message.
> 
> Does this also apply to variable names?

No, in code follow the existing style, variables are lower-case, `#define` are all upper.
